add docker image build pytorch-linux-xenial-cuda9.2-cudnn7-py3-gcc5.4

glaringlee opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#37610 add docker image build pytorch-linux-xenial-cuda9.2-cudnn7-py3-gcc5.4**

New docker image is already created here:
http://docker.pytorch.org/pytorch.html
tag is 9a3986fa-7ce7-4a36-a001-3c9bef9892e2

This will be used to create a new pytorch CI test under ubuntu16.04+cuda9.2+gcc5.4 which we were not covered.

Differential Revision: [D21336643](https://our.internmc.facebook.com/intern/diff/D21336643)

Commit:
add docker image build ubuntu16.04-cuda9.2-cudnn7-gcc5.4-py3.6

[ghstack-poisoned]

Commit:
Update on "add docker image build ubuntu16.04-cuda9.2-cudnn7-gcc5.4-py3.6"


Differential Revision: [D21336643](https://our.internmc.facebook.com/intern/diff/D21336643)

[ghstack-poisoned]

Commit:
Update on "add docker image build ubuntu16.04-cuda9.2-cudnn7-gcc5.4-py3.6"


Differential Revision: [D21336643](https://our.internmc.facebook.com/intern/diff/D21336643)

[ghstack-poisoned]

Commit:
Update on "add docker image build ubuntu16.04-cuda9.2-cudnn7-gcc5.4-py3.6"


Differential Revision: [D21336643](https://our.internmc.facebook.com/intern/diff/D21336643)

[ghstack-poisoned]

Commit:
Update on "add docker image build ubuntu16.04-cuda9.2-cudnn7-gcc5.4-py3.6"


Differential Revision: [D21336643](https://our.internmc.facebook.com/intern/diff/D21336643)

[ghstack-poisoned]

Commit:
Update on "add docker image build ubuntu16.04-cuda9.2-cudnn7-gcc5.4-py3.6"


Differential Revision: [D21336643](https://our.internmc.facebook.com/intern/diff/D21336643)

[ghstack-poisoned]

Commit:
Update on "add docker image build ubuntu16.04-cuda9.2-cudnn7-gcc5.4-py3.6"


Differential Revision: [D21336643](https://our.internmc.facebook.com/intern/diff/D21336643)

[ghstack-poisoned]

Commit:
Update on "add docker image build ubuntu16.04-cuda9.2-cudnn7-gcc5.4-py3.6"


Differential Revision: [D21336643](https://our.internmc.facebook.com/intern/diff/D21336643)

[ghstack-poisoned]

Commit:
Update on "add docker image build ubuntu16.04-cuda9.2-cudnn7-gcc5.4-py3.6"


Differential Revision: [D21336643](https://our.internmc.facebook.com/intern/diff/D21336643)

[ghstack-poisoned]

Commit:
Update on "add docker image build ubuntu16.04-cuda9.2-cudnn7-gcc5.4-py3.6"


New docker image is already created here:
http://docker.pytorch.org/pytorch.html
tag is 9a3986fa-7ce7-4a36-a001-3c9bef9892e2

Differential Revision: [D21336643](https://our.internmc.facebook.com/intern/diff/D21336643)

[ghstack-poisoned]

Commit:
Update on "add docker image build pytorch-linux-xenial-cuda9.2-cudnn7-py3-gcc5.4"


New docker image is already created here:
http://docker.pytorch.org/pytorch.html
tag is 9a3986fa-7ce7-4a36-a001-3c9bef9892e2

Differential Revision: [D21336643](https://our.internmc.facebook.com/intern/diff/D21336643)

[ghstack-poisoned]

seemethere commented on .circleci/docker/build.sh:
```diff
@@ -84,6 +84,15 @@ case "$image" in
+  pytorch-linux-xenial-cuda9.2-cudnn7-py3-gcc5.4)
+    CUDA_VERSION=9.2
+    CUDNN_VERSION=7
+    ANACONDA_PYTHON_VERSION=3.6
```
We should probably just standardized on using the latest version of Python wherever possible.

Can you update this to Python 3.8?

glaringlee replied:
Sure, will do

glaringlee replied:
@seemethere 
I tried to build python 3.8 image, it seems to me that there is no simple way at this point to create such docker image. 
There are 2 things that blocking it:

1. Pytorch docker image install llvmlite 0.30.0 which does not support llvm9 under python 3.8, it need llvm-config which is not available in llvm9. 

2. Yes, I tried switching to llvmlite 0.31.0 (or 32), it works, the docker image can be built, but the gcc-5 version is upgraded to 5.5 instead of 5.4, this will affect all the pytorch docker image that using gcc-5, and currently, all the gcc-5 pytorch CI test assume it uses gcc 5.4,  so I think it would be better to make such change in a separate issue since this is not trivial.

I'd like to keep using python3.6 for this image, please advise.

seemethere replied:
Sure let's stick to Python 3.6 here. Can you create a tracking issue to update these images to Python 3.8 with this information inside of the issue?

glaringlee replied:
sure. Created https://github.com/pytorch/pytorch/issues/37847
Will go back to 3.6 and request review agan.

Commit:
Update on "add docker image build pytorch-linux-xenial-cuda9.2-cudnn7-py3-gcc5.4"


New docker image is already created here:
http://docker.pytorch.org/pytorch.html
tag is 9a3986fa-7ce7-4a36-a001-3c9bef9892e2

This will be used to create a new pytorch CI test under ubuntu16.04+cuda9.2+gcc5.4 which we were not covered.

Differential Revision: [D21336643](https://our.internmc.facebook.com/intern/diff/D21336643)

[ghstack-poisoned]

Commit:
Update on "add docker image build pytorch-linux-xenial-cuda9.2-cudnn7-py3-gcc5.4"


New docker image is already created here:
http://docker.pytorch.org/pytorch.html
tag is 9a3986fa-7ce7-4a36-a001-3c9bef9892e2

This will be used to create a new pytorch CI test under ubuntu16.04+cuda9.2+gcc5.4 which we were not covered.

Differential Revision: [D21336643](https://our.internmc.facebook.com/intern/diff/D21336643)

[ghstack-poisoned]

ailzhang commented on .circleci/verbatim-sources/workflows-docker-builder.yml:
```diff
@@ -1,11 +1,4 @@
   docker_build:
-    triggers:
```
These changes need to be restored before merging.

glaringlee replied:
Yep, I was trying to switch to a python 3.8 image, will add them back later

ailzhang commented on .circleci/verbatim-sources/workflows-ecr-gc.yml:
```diff
@@ -1,11 +1,4 @@
   ecr_gc:
-    triggers:
```
And this one

seemethere reviewed (commented):
LGTM after we restore the triggers

Commit:
Update on "add docker image build pytorch-linux-xenial-cuda9.2-cudnn7-py3-gcc5.4"


New docker image is already created here:
http://docker.pytorch.org/pytorch.html
tag is 9a3986fa-7ce7-4a36-a001-3c9bef9892e2

This will be used to create a new pytorch CI test under ubuntu16.04+cuda9.2+gcc5.4 which we were not covered.

Differential Revision: [D21336643](https://our.internmc.facebook.com/intern/diff/D21336643)

[ghstack-poisoned]

Commit:
Update on "add docker image build pytorch-linux-xenial-cuda9.2-cudnn7-py3-gcc5.4"


New docker image is already created here:
http://docker.pytorch.org/pytorch.html
tag is 9a3986fa-7ce7-4a36-a001-3c9bef9892e2

This will be used to create a new pytorch CI test under ubuntu16.04+cuda9.2+gcc5.4 which we were not covered.

Differential Revision: [D21336643](https://our.internmc.facebook.com/intern/diff/D21336643)

[ghstack-poisoned]

glaringlee commented:
@seemethere @ailzhang 
restored to 3.6 and rebased master. please check and stamp if looks good. thx.

Commit:
Update on "add docker image build pytorch-linux-xenial-cuda9.2-cudnn7-py3-gcc5.4"


New docker image is already created here:
http://docker.pytorch.org/pytorch.html
tag is 9a3986fa-7ce7-4a36-a001-3c9bef9892e2

This will be used to create a new pytorch CI test under ubuntu16.04+cuda9.2+gcc5.4 which we were not covered.

Differential Revision: [D21336643](https://our.internmc.facebook.com/intern/diff/D21336643)

[ghstack-poisoned]

Commit:
Update on "add docker image build pytorch-linux-xenial-cuda9.2-cudnn7-py3-gcc5.4"


New docker image is already created here:
http://docker.pytorch.org/pytorch.html
tag is 9a3986fa-7ce7-4a36-a001-3c9bef9892e2

This will be used to create a new pytorch CI test under ubuntu16.04+cuda9.2+gcc5.4 which we were not covered.

Differential Revision: [D21336643](https://our.internmc.facebook.com/intern/diff/D21336643)

[ghstack-poisoned]

Commit:
Update on "add docker image build pytorch-linux-xenial-cuda9.2-cudnn7-py3-gcc5.4"


New docker image is already created here:
http://docker.pytorch.org/pytorch.html
tag is 9a3986fa-7ce7-4a36-a001-3c9bef9892e2

This will be used to create a new pytorch CI test under ubuntu16.04+cuda9.2+gcc5.4 which we were not covered.

Differential Revision: [D21336643](https://our.internmc.facebook.com/intern/diff/D21336643)

[ghstack-poisoned]

Commit:
Update on "add docker image build pytorch-linux-xenial-cuda9.2-cudnn7-py3-gcc5.4"


New docker image is already created here:
http://docker.pytorch.org/pytorch.html
tag is 9a3986fa-7ce7-4a36-a001-3c9bef9892e2

This will be used to create a new pytorch CI test under ubuntu16.04+cuda9.2+gcc5.4 which we were not covered.

Differential Revision: [D21336643](https://our.internmc.facebook.com/intern/diff/D21336643)

[ghstack-poisoned]

ailzhang reviewed (approved):
LGTM!
